[CRUSADE] Upgrade ruby and cleanup ➕

skrix opened the pull request:
### Description
Since there're a lot of abstractions and components, application complexity growth and the only God knows what's going on in production and how everything still works. The main goal of this pull request is to iteratively remove unused parts, increase the readability of code, decrease the number of used abstractions and provide simpler code structure. 

### How to test instructions
Changes will be pushed only to this branch for now and will be updated from time to time. After the point of the ready-to-use version, it will be deployed to the test server. Also, there will be a test suite provided ;)

### Reviews & Collaboration is welcomed and appreciated!

skrix committed:
[STEP-1] Remove unneeded abstractions :fire:
[STEP-1.1] Move model-only modules to model concerns
[STEP-1.2] Move single-used validation from validator to model
[STEP-1.3] Fix CI config :confetti_ball:
[STEP-1.4] Update ruby to 2.6.5 :wink:

skrix commented:
# Crusade Implementation Plan
| Current State   | Plan                                                                                              |
|-----------------|---------------------------------------------------------------------------------------------------|
| app/uploaders   | Move to one uploader service under models/store folder and use shrine as an uploader              |
| app/inputs      | ImagePreviewInput use shrine preview. DateTimeInput use datepicker gem or just customize in place |
| app/helpers     | Reduce using by implementing presenter's mixins or use cells and move under them                  |
| app/policies    | Replace with ez-permissions                                                                       |
| app/services    | Move to logic layer and group by responsibilities with the module as a facade                         |
| app/interactors | Replace with services under logic layer and group by responsibilities                             |
| app/mailer_previews | **_Need more investigation on the usage of this abstraction_**                                          |
| app/forms       | Move under models/forms to reach the incapsulation of persistence logic under models layer        |
| app/finders     | Move under models/queries and provide standard interface                                          |
| app/controllers | Reduce amount of responsibilities by moving logic under logic layer and prepare for migration to graphql |

### If someone have any suggestions or questions feel free to ask me :wink:

sveredyuk commented on .ruby-version:
```diff
@@ -1 +1 @@
-2.3.1
+2.6.3
```
Cool, but the current production server version is 2.3.1 ;)

sveredyuk reviewed (commented):
Well done. Let's review all this 🎄

denys-medynskyi commented:
In general, I appreciate 👏 your attitude and plan.

I have also some suggestions:

1. Clean up dead code / unused features first 
**Reasoning**: why we need to refactor code that is not needed? Let's clean up the mess first, and refactor it later

Article for reading: https://kevinjalbert.com/find-and-bury-dead-code/

2. Small PRs instead of 1 gigantic PR
**Reasoning**: it is easier to review, easier to revert, easier to debug issues.

3. Ask for more man-power
**Reasoning**: It is a big amount of work, it will be easier and more fun to do in a bigger group. Also, it can increase the number of active members

- Let's spread the word that we are going to clean up the website in our #genral slack.
- Let's do regular hackathons
- Give away for PRs?

4. Define real-world problems
**Reasoning**: we should focus on making the visitor experience and event admin experience.
I truly agree we need to clean up the code mess, however, we have more problems: *logic* mess.
For example, we have many event/user statuses that are not used.

denys-medynskyi commented:
https://devcenter.heroku.com/articles/git#deploying-from-a-branch-besides-master

denys-medynskyi commented:
`git push heroku refactoring:master`

denys-medynskyi commented:
I think we need to squash existing commits to 1.
It will simplify rebase.

Each refactoring branch should start with this branch.

denys-medynskyi committed:
[STEP-2] Remove unused features: Donations & Goals 🔥

* [C-870] Remove Donation
[STEP-1] Remove Donation related code
[STEP-2] Drop database table
[STEP-3] Remove specs
[STEP-4] Fix specs

* [C-868] Remove Goals feature :fire:
[STEP-1] Revert bundler version
[STEP-2] Drop Goal from database
[STEP-3] Adjust specs/features

skrix committed:
[STEP-3] Add improvements from #hacktoberfest contributors :bow:

* [STEP-1] Remove redundant lines in README

* [I-1] Remove the need for the Timecop gem
* [I-2] Remove duplicate pending migrations checks for testing
* [I-3] Remove unneeded dependency to the DatabaseCleaner
* [I-4] Fixed authentication with providers when email was changed
* [I-5] Make venue mandatory for an event :tada:

Co-authored-by: Filipe W. Lima <fwl.ufpe@gmail.com>
Co-authored-by: Alexandr Kovtunov <1sasha.sasha1@gmail.com>
Co-authored-by: aleksacastle <aaklochko@gmail.com>

skrix commented:
@deny7ko 
> I think we need to squash existing commits to 1.
> It will simplify rebase.

I squashed commits. Yep, it was a good idea. 

> Each refactoring branch should start with this branch.

Of course, next time don't forget this when will start working on a new issue ;) 
But we always have a great rebase 😉

skrix commented:
The live version of this branch: 
https://pivorak-staging.herokuapp.com/

* we will try to keep it up-to-date but sometimes expectations meet the reality, so ping if notice something outdated

denys-medynskyi committed:
[STEP-4] Remove not used functionality #1

* Remove become speaker
* Remove become speaker link and feature
- we use google form instead

andriivitiv committed:
[STEP-4.1] Fix typos in courses specs :duck:

skrix committed:
Fix specs

skrix committed:
Fix codeclimate
